workspaces/entity-feedback/plugins/entity-feedback/src/components/FeedbackResponseDialog/FeedbackResponseDialog.tsx

```diff
@@ -82,13 +107,16 @@ export const FeedbackResponseDialog = (props: FeedbackResponseDialogProps) => {
   const [responseSelections, setResponseSelections] = useState(
     Object.fromEntries(feedbackDialogResponses.map(r => [r.id, false])),
   );
-  const [comments, setComments] = useState('');
+  const [comments, setComments] = useState<Comments>({
+    responseComments: {},
+    additionalComments: '',
+  });
   const [consent, setConsent] = useState(true);
 
   const [{ loading: saving }, saveResponse] = useAsyncFn(async () => {
     try {
       await feedbackApi.recordResponse(stringifyEntityRef(entity), {
-        comments,
+        comments: JSON.stringify(comments),
         consent,
         response: Object.keys(responseSelections)
           .filter(id => responseSelections[id])
```

Comment:
I think we want to filter out comments from responses that might've been unselected right? eg. user checks "incorrect" selection, fills in some comments for that selection but then changes their mind and unchecks it before hitting submit

Reply:
Thanks for catching this! Updated 

https://github.com/backstage/community-plugins/assets/146387693/e39c68bb-d520-49a2-b96d-191aab74e627